sw/device/tests/edn_boot_mode.c

```diff
@@ -0,0 +1,170 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_csrng.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/dif/dif_entropy_src.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/edn_testutils.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/otbn_testutils.h"
+#include "sw/device/lib/testing/rv_core_ibex_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+#include "sw/device/tests/otbn_randomness_impl.h"
+
+#include "edn_regs.h"  // Generated
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+enum {
+  kEdnBootModeTimeout = (10 * 1000 * 1000),
+  kEdnBootModeOtbnRandomnessIterations = 1,
+};
+
+static dif_entropy_src_t entropy_src;
+static dif_csrng_t csrng;
+static dif_edn_t edn0;
+static dif_edn_t edn1;
+static dif_otbn_t otbn;
+static dif_rv_core_ibex_t rv_core_ibex;
+
+dif_entropy_src_config_t entropy_src_config = {
+    .fips_enable = false,
+    .route_to_firmware = false,
+    .bypass_conditioner = false,
+    .single_bit_mode = kDifEntropySrcSingleBitModeDisabled,
+    .health_test_window_size = 0x0200,
+    .alert_threshold = 2,
+};
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// Initializes the peripherals used in this test.
+static void init_peripherals(void) {
+  CHECK_DIF_OK(dif_entropy_src_init(
+      mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+  CHECK_DIF_OK(dif_csrng_init(
+      mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR), &edn0));
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR), &edn1));
+  CHECK_DIF_OK(
+      dif_otbn_init(mmio_region_from_addr(TOP_EARLGREY_OTBN_BASE_ADDR), &otbn));
+  CHECK_DIF_OK(dif_rv_core_ibex_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
+      &rv_core_ibex));
+}
+
+static void configure_otbn(void) {
+  otbn_randomness_test_prepare(&otbn, kEdnBootModeOtbnRandomnessIterations);
+}
+
+// configure the entropy complex
+static status_t entropy_config(unsigned int round) {
+  // EDN params with high values for glen so we don't have to reseed.
+  dif_edn_auto_params_t edn_params0 = edn_testutils_auto_params_build(
+      true, /*res_itval=*/32, /*glen_val=*/4095);
+  dif_edn_auto_params_t edn_params1 = edn_testutils_auto_params_build(
+      true, /*res_itval=*/32, /*glen_val=*/4095);
+  // Disable the entropy complex.
+  CHECK_STATUS_OK(entropy_testutils_stop_all());
```

Comment:
As this function returns `status_t`, consider replacing the `CHECK_STATUS_OK` by:
```suggestion
  TRY(entropy_testutils_stop_all());
```
Or maybe just make the function return `void`

Reply:
Thanks, I'll change this before the PR is merged.